source/server/hot_restarting_parent.cc

```diff
@@ -96,11 +96,7 @@ void HotRestartingParent::exportStatsToChild(HotRestartMessage::Reply::Stats* st
     (*stats->mutable_gauges())[gauge->name()] = gauge->value();
   }
   for (const auto& counter : server_->stats().counters()) {
-    uint64_t latched_delta = counter->latch();
-    (*stats->mutable_counter_deltas())[counter->name()] = latched_delta;
-    if (counters_already_sent_.insert(counter->nameCStr()).second) {
-      (*stats->mutable_counter_values())[counter->name()] = counter->value() - latched_delta;
-    }
+    (*stats->mutable_counter_deltas())[counter->name()] = counter->latch();
```

Comment:
Can you add a comment here around how it's safe that we are latching now because we have already stopped flushing stats, etc.? Maybe even ASSERT that somehow? Not sure what the best way would be.

Reply:
I left a comment. I do think it would be doable and nice to have an ASSERT, but a little clunky. The best way I can think of is to introduce another flag bit for "in hot restart transition", and a new second version of latch(), maybe latchHotRestart(). Normal latch() would assert that the flag is false, and latchHotRestart() would set it. Does that sound reasonable, or too much to be worth it?

Reply:
Yeah agreed not sure it's worth it. I think the comment is probably fine. @jmarantz any thoughts on this?

Reply:
I don't have enough insight into the dynamics yet to offer a concrete proposal on how to write the assert, so the comment LGTM for now.